provider/build.gradle.kts

```diff
@@ -52,6 +52,16 @@ val clean: Delete by tasks
 clean.delete(apiExtensionsOutputDir)
 
 
+// -- Version manifest properties --------------------------------------
+val versionsManifestOutputDir = file("$buildDir/versionsManifest")
+val writeVersionsManifest by tasks.creating(WriteProperties::class) {
+    outputFile = versionsManifestOutputDir.resolve("gradle-kotlin-dsl-versions.properties")
+    property("provider", version)
+    property("kotlin", kotlinVersion)
+}
```

Comment:
Why aren't you using the `ProcessResources` task instead and just embedding this as a resource in the `provider.jar`?

Reply:
The effect is the same but you're right, using `ProcessResources` is simpler, fixed.

Reply:
Not quite what I meant, but I guess this works.
I was expecting you to put a `gradle-kotlin-dsl-versions.properties` in the `src/resources` directory and then use the `ProcessResources` task to replace some things with the versions.